pkg/pool-hooks/contracts/MevHook.sol

```diff
@@ -262,6 +248,37 @@ contract MevHook is BaseHooks, SingletonAuthentication, VaultGuard, IMevHook {
         _setPoolMevTaxThreshold(pool, newPoolMevTaxThreshold);
     }
 
+    /*******************************************************
+                        Helper functions
+    *******************************************************/
+
+    function _calculateSwapFeePercentage(
+        uint256 staticSwapFeePercentage,
+        uint256 multiplier,
+        uint256 threshold
+    ) internal view returns (uint256) {
+        // If gasprice is lower than basefee, the transaction is invalid and won't be processed. Gasprice is set
+        // by the transaction sender, is always bigger than basefee, and the difference between gasprice and basefee
+        // defines the priority gas price (the part of the gas cost that will be paid to the validator).
+        uint256 priorityGasPrice = _getPriorityGasPrice();
+        uint256 maxMevSwapFeePercentage = _maxMevSwapFeePercentage;
+
+        // If `priorityGasPrice` < threshold, this indicates the transaction is from a retail user, so we should not
+        // impose the MEV tax. Also, if mev fee cap is lower than static fee percentage, returns the static.
+        if (priorityGasPrice < threshold || maxMevSwapFeePercentage < staticSwapFeePercentage) {
+            return staticSwapFeePercentage;
+        }
+
+        uint256 mevSwapFeePercentage = staticSwapFeePercentage + (priorityGasPrice - threshold).mulDown(multiplier);
```

Comment:
```suggestion
        // At this point, `priorityGasPrice >= threshold` and `maxMevSwapFeePercentage >= staticSwapFeePercentage`.
        uint256 mevSwapFeePercentage = staticSwapFeePercentage + (priorityGasPrice - threshold).mulDown(multiplier);
```

Are we going to include `tryMul` in a separate PR then?

Reply:
I just included tryMul.